Bio/Seq.py

```diff
@@ -1375,6 +1440,82 @@ def count(self, sub, start=0, end=sys.maxsize):
             else:
                 return 0
 
+    def count_overlap(self, sub, start=0, end=sys.maxsize):
+        """Return an overlapping count.
+
+        For a non-overlapping search use the count() method.
+
+        Returns an integer, the number of occurrences of substring
+        argument sub in the (sub)sequence given by [start:end].
+        Optional arguments start and end are interpreted as in slice
+        notation.
+
+        Arguments:
+         - sub - a string or another Seq object to look for
+         - start - optional integer, slice start
+         - end - optional integer, slice end
+
+        e.g.
+
+        >>> UnknownSeq(4, character="N").count_overlap("NN")
+        3
+        >>> UnknownSeq(4, character="N").count_overlap("NNN")
+        2
+        >>> UnknownSeq(7, character="N").count_overlap("NN")
+        6
+        >>> UnknownSeq(7, character="N").count_overlap("NNN")
+        5
+        >>> UnknownSeq(7, character="N").count_overlap("NN", 1, 5)
+        3
+        >>> UnknownSeq(7, character="N").count_overlap("NN", -5, -1)
+        3
+
+        Where substrings do not overlap, should behave the same as
+        the count() method:
+
+        >>> UnknownSeq(4, character="N").count_overlap("N")
+        4
+        >>> UnknownSeq(4, character="N").count_overlap("N") == UnknownSeq(4, character="N").count("N")
+        True
+        >>> UnknownSeq(4, character="N").count_overlap("A")
+        0
+        >>> UnknownSeq(4, character="N").count_overlap("A") == UnknownSeq(4, character="N").count("A")
+        True
+        >>> UnknownSeq(4, character="N").count_overlap("AA")
+        0
+        >>> UnknownSeq(4, character="N").count_overlap("AA") == UnknownSeq(4, character="N").count("AA")
+        True
+
+        HOWEVER, do not use this method for such cases because the
+        count() method is much for efficient.
+        """
+        sub_str = self._get_seq_str_and_check_alphabet(sub)
+        if len(sub_str) == 1:
+            if str(sub_str) == self._character:
+                if start == 0 and end >= self._length:
+                    return self._length
+                else:
+                    # This could be done more cleverly...
+                    return str(self).count(sub_str, start, end)
+            else:
+                return 0
+        else:
+            if set(sub_str) == set(self._character):
+                if start == 0 and end >= self._length:
+                    return self._length - len(sub_str) + 1
+                else:
+                    # This could be done more cleverly...
+                    self = str(self)
```

Comment:
Please avoid redefining ``self``; see also separate comment about how to implement this efficiently